server/datastore/mysql/migrations/tables/20240320145650_UpdateDEPProfilesToAwaitDeviceConfigured.go

```diff
@@ -0,0 +1,64 @@
+package tables
+
+import (
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"time"
+)
+
+func init() {
+	MigrationClient.AddMigration(Up_20240320145650, Down_20240320145650)
+}
+
+func Up_20240320145650(tx *sql.Tx) error {
+	// This migration is to re-generate and re-register with Apple the DEP
+	// enrollment profile(s) so that await_device_configured is set to true.
+	// We do this by doing the equivalent of:
+	//
+	// 	 worker.QueueMacosSetupAssistantJob(ctx, ds, logger,
+	//     worker.MacosSetupAssistantUpdateAllProfiles, nil)
+	//
+	// but without calling that function, in case the code changes in the future,
+	// breaking this migration. Instead we insert directly the job in the
+	// database, and the worker will process it shortly after Fleet restarts.
```

Comment:
@roperzh do you know if that means that existing DEP-enrolled devices will now need to be explicitly released? The Apple docs mention it blocks in Setup Assistant so presumably devices that are already past this assistant will be ok, but worth double-checking in tests? Because re-generating the profiles also re-assigns them to existing hosts here: https://github.com/fleetdm/fleet/blob/main/server/worker/macos_setup_assistant.go#L113-L137

Reply:
Yes, I believe we're fine here! those settings only affect the next time the device goes through the setup assistant 👍